pandas/core/dtypes/cast.py

```diff
@@ -712,7 +712,8 @@ def astype_nansafe(arr, dtype, copy=True):
     elif is_object_dtype(arr):
 
         # work around NumPy brokenness, #1987
-        if np.issubdtype(dtype.type, np.integer):
+        is_time = is_timedelta64_dtype(dtype)
+        if np.issubdtype(dtype.type, np.integer) and is_time:
```

Comment:
use is_integer_dtype and also add a comment here (and can remove the existing work around comment)

Reply:
Ok, but it breaks:
https://github.com/pandas-dev/pandas/blob/f76a3f352a0d32f0fb42959e38fe80669add6822/pandas/tests/reshape/merge/test_merge.py#L531-L538
Because of:
https://github.com/pandas-dev/pandas/blob/f76a3f352a0d32f0fb42959e38fe80669add6822/pandas/core/dtypes/cast.py#L688-L697
Are this necessary? Here timedelta64 are turned to float64.
Should I change the test or this part of `astype_nansafe`?
Also, I've notices that conversions like `pd.Series([None, 1, 2]).astype('timedelta64')` do not work now. Should I open a new issue?

Reply:
Might be missing something but shouldn't it not be going down this branch? i.e. isn't the statement `dtype not in [_INT64_DTYPE, _TD_DTYPE]` supposed to be false?

Reply:
Is IntNADType handled correctly here?

Reply:
@WillAyd yah at this point I’m not sure if it’s tangential, but it looks like the TDDTYPE check should go above the (PY3 ...) check in line 688, after which that check is always-true.